### DISCUSSION TITLE
Welcome to kohii Discussions!

### DISCUSSION THREAD

**Discussion by eneim** [eneim/kohii#121] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by nicolasmartino-dev** [comment]
Hi eneim, I have a question about kohii. Would it be possible to use it just in order to play audio?
I would like for instance to pass a PlayerControlView instead of a PlayerView and always show the controls.
Another strategy would maybe to have the PlayerView with surface none and have a way to tell kohii not to hide the controls?

Also I used your tutorial in order to display the controls: https://www.notion.so/Manual-playback-controller-in-Kohii-212927dd75de4971bc28d27e4b34911a
But then I wanted to use `app:controller_layout_id="@layout/audio_custom_controls"` in order to display my own override of exoPlayer controls like:
```
<com.google.android.exoplayer2.ui.PlayerView
                    android:id="@+id/audio_player_control_view"
                    android:layout_width="0dp"
                    android:layout_height="0dp"
                    app:controller_layout_id="@layout/audio_custom_controls"
                    app:hide_on_touch="false"
                    app:layout_constraintBottom_toBottomOf="parent"
                    app:layout_constraintEnd_toEndOf="parent"
                    app:layout_constraintStart_toStartOf="parent"
                    app:layout_constraintTop_toTopOf="parent"
                    app:show_timeout="-1" />
```
unfortunately it does not display well and play button doesn't work (yet)

In your tutorial you say: 

> There will be more ways for your to customize the controller UI and behavior, but it is out of the scope of this tutorial.

Do you have any example?

Thank you

**Reply by eneim** [reply]
> Another strategy would maybe to have the PlayerView with surface none

This is a good first step I think. About the controller, you may need to have a `controller` in your setup. I will try a quick demo and update this thread later.

**Reply by nicolasmartino-dev** [reply]
Here is my controller:
```
controller = object : Controller {
                override fun kohiiCanPause(): Boolean = true
                override fun kohiiCanStart(): Boolean = false

                override fun setupRenderer(
                    playback: Playback,
                    renderer: Any?
                ) {
                    pause(playback)
                    if (renderer is PlayerView) {
                        renderer.useController = true
                        renderer.setControlDispatcher(kohii.createControlDispatcher(playback))
                        renderer.showController()
                    }
                }

                override fun teardownRenderer(
                    playback: Playback,
                    renderer: Any?
                ) {
                    if (renderer is PlayerView) {
                        renderer.useController = false
                        renderer.setControlDispatcher(null)
                    }
                }
            }
```

**Reply by nicolasmartino-dev** [reply]
I managed to display an audio player but I have multiple players in my recyclerview and the others have their controls hidden. It only shows the controls when the player has the focus (the first viewable on top of the recyclerview). Do you know how I could handle this? Thank you

**Reply by nicolasmartino-dev** [reply]
Hi @eneim, I changed a bit what I did in order to make it look like what you did in your sample Kohii  + Videos with manual controller. Now I have the play pause working. Remains two challenges for me:
- Sometimes in my recyclerView the play button is not triggering for some items (maybe a problem related to the recyclerView)
- Do you know how I can display the exo_position, exo_progress and exo_duration. Should I update them manually via the manager ?
